/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package tflite.task.processor;

option java_multiple_files = true;
option java_package = "org.tensorflow.lite.task.processor.proto";

// A single nearest neighbor.
// Next ID: 3.
message NearestNeighbor {
  // User-defined metadata about the result. This could be a label, a unique ID,
  // a serialized proto of some sort, etc.
  optional bytes metadata = 1;

  // The distance score indicating how confident the result is. Lower is better.
  optional float distance = 2;
}

// Results from a search as a list of nearest neigbors.
// Next ID: 2.
message SearchResult {
  // The nearest neighbors, sorted by increasing distance order.
  repeated NearestNeighbor nearest_neighbors = 1;
}
